tests/phpunit/tests/rest-api/rest-pattern-directory-controller.php

```diff
@@ -399,6 +410,85 @@ public function test_get_item_schema() {
 		$this->markTestSkipped( "The controller's schema is hardcoded, so tests would not be meaningful." );
 	}
 
+	/**
+	 * Tests if the transient key gets generated correctly.
+	 *
+	 * @dataProvider data_get_query_parameters
+	 *
+	 * @covers WP_REST_Pattern_Directory_Controller::get_transient_key
+	 *
+	 * @since 6.0.0
+	 *
+	 * @ticket 55617
+	 *
+	 * @param array     $parameters_1   Expected query arguments.
+	 * @param array     $parameters_2   Actual query arguments.
+	 * @param string    $message        An error message to display.
+	 * @param bool      $assert_same    Assertion type (assertSame vs assertNotSame).
+	 */
+	public function test_transient_keys_get_generated_correctly( $parameters_1, $parameters_2, $message, $assert_same = true ) {
+		$method = $this->get_reflection_method( self::$controller, 'get_transient_key' );
```

Comment:
```suggestion
                 $controller = new WP_REST_Pattern_Directory_Controller();
		$method = $this->get_reflection_method( $controller, 'get_transient_key' );
```

Reply:
Refactored in 7d1c51221c42336c151aa454c7ce3b407b7e250f